core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -39,15 +38,12 @@
 	@Autowired
 	private AnalysisEngineCacheService cacheService;
 	@Autowired
-	private UserService userService;
-	@Autowired
 	private UserAnonymizedService userAnonymizedService;
 	@Autowired
 	private MessageService messageService;
 
-	public void analyze(UUID userID, AppActivityDTO[] appActivities)
+	public void analyze(UUID userAnonymizedID, AppActivityDTO[] appActivities)
 	{
-		UUID userAnonymizedID = userService.getPrivateUser(userID).getPrivateData().getUserAnonymizedID();
 		UserAnonymizedDTO userAnonymized = userAnonymizedService.getUserAnonymized(userAnonymizedID);
 		for (AppActivityDTO appActivity : appActivities)
```

Comment:
Even then, I'm afraid the current code doesn't accept app activity reports that are reported way after they happened, while network (Smoothwall) activities have bene reported in that same period. What do you think?